internal/discharger: discharge token unique ID

mhilton opened the pull request:
Use the unique and immutable ProviderID in new discharge tokens. Old
username-based discharge tokens will still work.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/48/

mhilton commented on cmd/candid/internal/admincmd/command.go:
```diff
@@ -86,15 +86,24 @@ func (c *candidCommand) Client(ctxt *cmd.Context) (*candidclient.Client, error)
+	if authInfo != nil {
+		// Agent authentication has been specified, so we probably don't
+		// want to use existing cookies (which might be logged in as a different
+		// user) or to fall back to interactive authentication.
```
I'm not sure how true that is, what if the auth info in the environment is for a different service?

rogpeppe replied:
I think it's more true than not - we can add another flag if need be.

rogpeppe commented:
@fabricematrat I'll address the docs issue in the next PR, where I'm moving the common flags into the supercommand.

jujugui commented:
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

mhilton committed:
internal/discharger: discharge token unique ID

Use the unique and immutable ProviderID in new discharge tokens. Old
username-based discharge tokens will still work.
